Community/Tdarr_Plugin_vdka_Tiered_CPU_CRF_Based_Configurable.js

```diff
@@ -0,0 +1,150 @@
+        name: 'bframe',
+        tooltip: `Specify amount of b-frames to use, 0-16.
+        
+        \\nExample:\\n
+        8`,
+      },
+      {
+        name: 'ffmpegPreset',
+        tooltip: `OPTIONAL, DEFAULTS TO SLOW IF NOT SET 
+        \\n Enter the ffmpeg preset you want, leave blank for default (slow) 
+        
+        \\nExample:\\n 
+          slow  
+        
+        \\nExample:\\n 
+          medium  
+        
+        \\nExample:\\n 
+          fast  
+        
+        \\nExample:\\n 
+          veryfast`,
+      },
+    ],
+  };
+}
+
+module.exports.plugin = function plugin(file, librarySettings, inputs) {
+  let crf = '';
+  // default values that will be returned
+  const response = {
+    processFile: true,
+    preset: '',
+    container: '.mkv',
+    handBrakeMode: false,
+    FFmpegMode: true,
+    reQueueAfter: true,
+    infoLog: '',
+  };
+
+  // check if the file is a video, if not the function will be stopped immediately
+  if (file.fileMedium !== 'video') {
+    response.processFile = false;
+    response.infoLog += '☒File is not a video! \n';
+    return response;
+  }
+  response.infoLog += '☑File is a video! \n';
+
+  // check if the file is already hevc
+  // it will not be transcoded if true and the plugin will be stopped immediately
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    if (file.ffProbeData.streams[i].codec_name.toLowerCase() === 'hevc') {
+      response.processFile = false;
+      response.infoLog += '☑File is already in hevc! \n';
+      break;
+    }
+  }
+
+  // if we made it to this point it is safe to assume there is no hevc stream
+  response.infoLog += '☒File is not hevc!\n';
+
+  // Check if preset is configured, default to slow if not
+  let ffmpegPreset = 'slow';
+  if (inputs.ffmpegPreset === undefined) {
+    response.infoLog += '☑Preset not set, defaulting to slow\n';
+  } else {
+    ffmpegPreset = `${inputs.ffmpegPreset}`;
+    response.infoLog += `☑Preset set as ${inputs.ffmpegPreset}\n`;
+  }
+
+  // set crf by resolution
+  switch (file.video_resolution) {
+    case '480p':
+    case '576p':
+      crf = inputs.sdCRF;
+      break;
+    case '720p':
+      crf = inputs.hdCRF;
+      break;
+    case '1080p':
+      crf = inputs.fullhdCRF;
+      break;
+    case '4KUHD':
+      crf = inputs.uhdCRF;
+      break;
+    default:
+      response.infoLog += 'Could for some reason not detect resolution, plugin will not proceed. \n';
+      response.processFile = false;
+      return response;
+  }
+
+  // encoding settings
+  response.preset += `,-map 0 -dn -c:v libx265 -preset ${ffmpegPreset}`
+  + ` -x265-params crf=${crf}:bframes=${inputs.bframe}:rc-lookahead=32:ref=6:b-intra=1:aq-mode=3`
+  + ' -a53cc 0 -c:a copy -c:s copy -max_muxing_queue_size 9999';
+  response.infoLog += `☑File is ${file.video_resolution}, using CRF value of ${crf}!\n`;
+  response.infoLog += 'File is being transcoded!\n';
+
+  return response;
+};
+module.exports.details = details;
```

Comment:
Either do it for the plugin too, or move this to line 1 pls

Reply:
> Either do it for the plugin too, or move this to line 1 pls

Done